src/Fixture/FixtureCreator.php

```diff
@@ -358,25 +358,37 @@ public function setSymlinkAll(bool $symlink_all): void {
   }
 
   /**
-   * Creates a BLT project.
+   * Creates a project.
    */
-  private function createBltProject(): void {
-    $this->output->section('Creating BLT project');
+  private function createProject(): void {
+    $this->output->section('Creating project');
     $stability_flag = '--stability=alpha';
-    if (($this->isDev || ($this->sut && $this->sut->getPackageName() === 'acquia/blt'))) {
+    if (($this->isDev || ($this->sut && ($this->sut->getPackageName() === 'acquia/blt' || $this->sut->getPackageName() === 'acquia/drupal-recommended-project')))) {
       $stability_flag = '--stability=dev';
     }
+    // Handle D8 fixtures (default).
     $version = ($this->isDev)
       ? $this->blt->getVersionDev($this->drupalCoreVersion)
       : $this->blt->getVersionRecommended($this->drupalCoreVersion);
+    $project_template = 'acquia/blt-project';
+    // Handle D9 differently, since BLT Project no longer exists.
+    if (version_compare($this->getResolvedDrupalCoreVersion(), 9, '>=')) {
+      $project_template = 'acquia/drupal-recommended-project';
+      $version = ($this->isDev) ? 'dev-master' : '^1';
+    }
+    // Handle the very special case of ORCA running as part of acquia/drupal-recommended-project builds.
+    if (getenv("ORCA_SUT_NAME") === 'acquia/drupal-recommended-project') {
+      // Have to use some unique branch name or Composer won't check out the path repo.
+      $version = 'dev-' . getenv("ORCA_SUT_BRANCH");
```

Comment:
I know this is ugly as sin, but I don't know what else to do. The problem is that Composer will _never_ utilize the path repo if the branch name (dev-master in the case of drupal-recommended-project) exists upstream somewhere. I have no idea why. I suspect that the branch that ORCA creates in `before_install.sh` on Travis does not have the full Git history, so Composer doesn't see it as being related to the upstream version.

I'm also not sure why this works for SUTs but not for fixtures (BLT Project had the same problem). With BLT, ORCA runs `composer require acquia/blt:12.9999.9999.999-dev` when `ORCA_SUT_BRANCH` is `12.x`. How the heck does that work?

Reply:
Per Travis on Zoom, this version string is actually pulled from Composer's guts by ORCA.